docs/cms/input-types.adoc

```diff
@@ -91,22 +91,50 @@ image::Content-selector-default.png[An input box with a dropdown options below i
   </config>
 </input>
 ----
-<1> *allowContentType* is used to limit the content types that may be selected using the <<allowContentType Samples>>
+<1> *allowContentType* is used to limit the content types that may be selected using the <<allowContentType>>
 <2> *allowPath* is used to limit the path of the content that may be selected using the <<allowPath Samples>>. By default, Content Selector
 only displays content from the same site.
 <3> *treeMode* toggles between tree and flat modes. Default value is `false` (flat mode). +
 image:Content-selector-tree.png[Input box with a dropdown with a tree structure, 350]
 <4> *hideToggleIcon* hides the toggle icon when set to `false`. +
 image:Content-selector-no-icon.png[Input box without an icon, 500]
 
-=== allowContentType Samples
+[#allowContentType]
+=== Allow content type pattern
+XP uses one of two modes when matching content type patterns:
 
+`LEGACY`:: this mode simulates the pattern matching functionality that was the default before XP 7.7.0. image:../images/xp-770.svg[XP 7.7.0,opts=inline] 
+`MATCH`:: this mode is the current default mode and has a few notable changes from `LEGACY` mode.
+
+The modes are explained in further detail below. You can configure which mode to use in  <<../deployment/config.adoc#admin, the admin config>>.
+==== Legacy +
+
+If the pattern value can be found in any part of the content type name, then it is considered a match.
+The pattern will be prefixed by the current application name unless:
+
+- the pattern uses the <application-name:content-name> format
+- the pattern includes one of `:` or `*`
+- the pattern starts with `${app}`.
+
+==== Match +
+A pattern will be treated as Java Regular Expression (with the exception to Special handling) and the whole content type name must match it to be accepted.
+The pattern will be prefixed by the current application name if it contains none of the following characters `\^$.|?*+()[{:`.
+
+==== Special handling
+
+.`${app}` template expression
+Most content type names start with the name of the application where they are stored. It may be quite messy and hard to maintain the whole application name in every pattern, so the `${app}` template expression can be used to replace it.
+
+.`*` character
+There is a special handling of `*` symbol which differs from usual regex handling. It will be considered by parser as '0 or more of any character except linebreaks' meanwhile regex treat it as '1 or more of the preceding token'.
```

Comment:
I wonder whether clarifying this with some examples would be good? Something like this? 🤓 

```suggestion
.The '*' character
In most regex engines, the `*` character means "0 or more of the preceding token". However, XP treats it differently and interprets it as "0 or more of any character(s) except for line breaks". In that way, it's similar to https://en.wikipedia.org/wiki/Glob_%28programming%29[glob patterns].

To further clarify, in XP:

- `*day` would match "monday", "day", and "mayday". This pattern would be invalid in most regex engines.
- `p*age` would match "page" and "package", but _not_ "age".
- `monk*` would match "monkey" and  "monk", but _not_ "mon".
```

Also, `*` usually means 0 or more of the preceding token, not 1 or more (at least in my experience).